chore(dependencies): Update go/k8s dependencies and ci

niladrih opened the pull request:
- update manusa/actions-setup-minikube to v2.7.1
- update minikube version to v1.26.1
- update kubernetes version to v1.25.0
- update minikube setup to use cilium cni
- update minikube setup to avoid storage provisioner plugin installation
- update k8s libraries to v0.25.4 (k8s v1.25.4)
- remove the vendor directory
- update go modules -- sigs.k8s.io/controller-runtime, k8s.io/klog

Signed-off-by: Niladri Halder <niladri.halder26@gmail.com>

akhilerm committed:
update go version to 1.19.3

- update go.mod / go.sum
- update vendor directory
- update build base image with go 1.19.3

Signed-off-by: Akhil Mohan <akhilerm@gmail.com>
Signed-off-by: Niladri Halder <niladri.halder26@gmail.com>

akhilerm committed:
Update changelogs

Signed-off-by: Akhil Mohan <akhilerm@gmail.com>
Signed-off-by: Niladri Halder <niladri.halder26@gmail.com>

niladrih committed:
chore(ci): update minikube setup
- update manusa/actions-setup-minikube to v2.7.1
- update minikube version to v1.26.1
- update kubernetes version to v1.25.0
- update minikube setup to use cilium cni
- update minikube setup to avoid storage provisioner plugin installation

Signed-off-by: Niladri Halder <niladri.halder26@gmail.com>

akhilerm commented on .github/workflows/build.yml:
```diff
@@ -347,10 +347,12 @@ jobs:
-          minikube version: v1.15.0
-          kubernetes version: 1.18.0
+          minikube version: 'v1.26.1'
+          kubernetes version: 'v1.25.0'
```
Along with this should we update the k8s dependecies in our code. Currently its at k8s 1.20. The controller runtime also would have to be updated to a matching version for the k8s version.

niladrih replied:
Updated the libraries.

akhilerm reviewed (commented):
Can you check why the integration tests are failing. May be due to mismatching k8s version. But not sure.

niladrih committed:
chore(dependencies): update k8s dependencies to v1.25.4

Signed-off-by: Niladri Halder <niladri.halder26@gmail.com>

akhilerm commented on go.mod:
```diff
@@ -25,3 +25,31 @@ require (
+
+replace k8s.io/cli-runtime => k8s.io/cli-runtime v0.25.4
+
+replace k8s.io/client-go => k8s.io/client-go v0.25.4
```
Instead of using replae directive here, this should be added in the require itself

niladrih replied:
✔️

niladrih committed:
chore(dependencies): update go modules
- update sigs.k8s.io/controller-runtime to v0.13.1
- update k8s.io/klog to k8s.io/klog/v2 v2.80.1

Signed-off-by: Niladri Halder <niladri.halder26@gmail.com>

niladrih committed:
chore(ci): use 'none' minikube driver

Signed-off-by: Niladri Halder <niladri.halder26@gmail.com>

akhilerm commented on .github/workflows/build.yml:
```diff
@@ -54,6 +54,11 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v2
 
+      - name: Set up Go v1.19
```
How was this working earlier without setting up go?

niladrih replied:
It was using the bundled Go version. Deliberately setting up the Go version, keeps things predictable.
https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md#go

akhilerm commented on .github/workflows/pull_request.yml:
```diff
@@ -53,6 +53,11 @@ jobs:
+        uses: actions/setup-go@v3
+        with:
+          go-version: '1.19.3'
+
```
Same question on how was this working earlier.

niladrih replied:
Refer comment reply above.

akhilerm commented on pkg/smart/doc.go:
```diff
@@ -55,29 +55,30 @@ An example usage can be like this -
-	if err != nil {
-		klog.Fatal(err)
-	}
+	func main() {
```
Why is this formatting needed?

niladrih replied:
It's the IDE. I'll fix this

niladrih replied:
I think I got all of it. Comment formatting reverted.

akhilerm reviewed (commented):
Can you check the formatting on the .go files once. Seems like its off.

niladrih committed:
chore(fmt): undo formatting

Signed-off-by: Niladri Halder <niladri.halder26@gmail.com>

akhilerm reviewed (approved):
LGTM
